File: PYTHON/Longest-Substring-Without-Repeating-Characters.py
1class Solution(object):
2    def lengthOfLongestSubstring(self, s):
3        left = count = 0
4        string= set()
5        for right in range(len(s)):
6            while s[right] in string:
7                string.remove(s[left])
8                left+=1
9            string.add(s[right])
10            count= max(count,right-left+1)
11        return count
12
13
14
15
16        

File: JAVA/Longest-Substring-Without-Repeating-Characters.java
1class Solution {
2    public int lengthOfLongestSubstring(String s) {
3       int left = 0, count = 0;
4       Set<Character> string = new HashSet<>();
5
6       for(int right=0;right<s.length();right++){
7        while(string.contains(s.charAt(right))){
8            string.remove(s.charAt(left));
9            left++;
10        }
11        count = Math.max(count,right-left+1);
12        string.add(s.charAt(right));
13       }
14
15       return count;
16    }
17}